arch/x86/adler32_avx512.c

```diff
@@ -33,52 +71,54 @@ Z_INTERNAL uint32_t adler32_avx512(uint32_t adler, const unsigned char *buf, siz
     /* in case short lengths are provided, keep it somewhat fast */
     if (UNLIKELY(len < 16))
         return adler32_len_16(adler, buf, len, sum2);
+#endif
 
-    const __mmask16 vs_mask = 1U << 15;
-    __m512i vs1 = _mm512_maskz_set1_epi32(vs_mask, adler);
-    __m512i vs2 = _mm512_maskz_set1_epi32(vs_mask, sum2);
+    __m512i vs1 = _mm512_zextsi128_si512(_mm_cvtsi32_si128(adler));
+    __m512i vs2 = _mm512_zextsi128_si512(_mm_cvtsi32_si128(sum2));
 
-    const __m512i dot1v = _mm512_set1_epi8(1);
     const __m512i dot2v = _mm512_set_epi8(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
                                           20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37,
                                           38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55,
                                           56, 57, 58, 59, 60, 61, 62, 63, 64);
     const __m512i dot3v = _mm512_set1_epi16(1);
+    const __m512i zero = _mm512_setzero_si512();
 
     while (len >= 64) {
-       __m512i vs1_0 = vs1;
```

Comment:
Looks like there are a lot of whitespace changes. Github doesn't show whether this is tab or space, but we use 4-space indentation, hope you make sure that is correct.

Reply:
The first underscore should align with `e`, not `l`... Not sure if it's rendering glitch or not...

Reply:
![spacing](https://user-images.githubusercontent.com/3620277/149960561-211ce986-19af-4b3c-a62b-cba213b2d90e.png)

Reply:
Sometimes I look at the actual file, not the diff to see everything is OK... I've seen many diffs render whitespace incorrectly especially when TABs and SPACEs are mixed in same text block.

Reply:
That is the actual file.  My vimrc doesn't place tabs and I've only ever edited that file on this system.  
```
set expandtab
set tabstop=4
set shiftwidth=4
set smarttab
set nocp
```
`:retab` makes no difference, they're definitely spaces if they weren't already.  In this case the diff is showing where I fixed what the indentation was, as I think that bit managed to get copied from the previous adler32 implementations.

Reply:
Here's the patient zero commit: https://github.com/zlib-ng/zlib-ng/blame/develop/arch/x86/adler32_ssse3.c#L54

Reply:
I guess the SSSE3 version should be cleaned up in separate pull request...